pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -249,6 +249,7 @@ func (ssr *Snapshotter) takeFullSnapshot() error {
 
 		metrics.LatestSnapshotRevision.With(prometheus.Labels{metrics.LabelKind: ssr.prevSnapshot.Kind}).Set(float64(ssr.prevSnapshot.LastRevision))
 		metrics.LatestSnapshotTimestamp.With(prometheus.Labels{metrics.LabelKind: ssr.prevSnapshot.Kind}).Set(float64(ssr.prevSnapshot.CreatedOn.Unix()))
+		metrics.SnapshotRequired.With(prometheus.Labels{metrics.LabelKind: snapstore.SnapshotKindFull}).Set(0)
 
```

Comment:
Shouldn't we also reset delta snapshot required metrics here?

Reply:
You're right. Thanks for the catch

Reply:
Actually move it out of after `else` block. It should be reset after both `if` and `else` path.

Reply:
Not necessary, because full snapshot required will be 0 if there are no revision changes. So there's no need to set it to 0 again in the `if` case.